packages/framework/src/Model/Mail/EventListener/EnvelopeListener.php

```diff
@@ -48,8 +47,7 @@ public function onMessage(MessageEvent $event): void
         }
 
         if ($allowedRecipients === []) {
-            // set a non-existing address because recipient list cannot be empty
-            $allowedRecipients = [new Address('no-reply@domain.tld')];
+            $event->reject();
```

Comment:
I think you need to add `return;` here, because `$event->getEnvelope()->setRecipients($allowedRecipients);` then still fails (recipients can not be empty), see the failing tests please